src/Field/FieldTypePluginManager.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Retrofit\Drupal\Field;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FieldTypePluginManager as CoreFieldTypePluginManager;
+use Retrofit\Drupal\Plugin\Field\FieldType\DecoratedFieldItem;
+
+final class FieldTypePluginManager extends CoreFieldTypePluginManager
+{
+    public function createFieldItem(FieldItemListInterface $items, $index, $values = null)
+    {
+        return new DecoratedFieldItem(parent::createFieldItem($items, $index, $values));
```

Comment:
This is crashing my site because it hides methods provided by field type classes.

Reply:
Shoot, thanks. It wasn't tested they way and is missing `__call`delegation

I still am unsure about it overall.. but it'd make a lot of existing D7 code work with fetching field values as an array

Reply:
This is teaching me a lot. I didn't know how to use the `__call` method for object decoration.

Reply:
Yeah! I just pushed a fix. You use `call_user_func_array`. I need to expand my test coverage. This class is a bit of a danger to add, but needed for interacting with fields.